### PR TITLE
Vts code cleanup

### PR DESCRIPTION
**Pull request by lmalenfant**

### CODE REVIEW

File: src/Vts.Test/MonteCarlo/Extensions/DatabaseExtensionMethodsTests.cs [hunk 1]
```diff
@@ -15,14 +16,20 @@ public void Validate_IspMCDatabase_returns_correct_values()
...
+            Assert.IsTrue(databaseType.IspMCDatabase());
             // validate those that are false
             databaseType = DatabaseType.DiffuseReflectance;
             Assert.IsFalse(databaseType.IspMCDatabase());
             databaseType = DatabaseType.DiffuseTransmittance;
             Assert.IsFalse(databaseType.IspMCDatabase());
             databaseType = DatabaseType.SpecularReflectance;
             Assert.IsFalse(databaseType.IspMCDatabase());
+            // check if enum set to something out of range
+            databaseType = (DatabaseType)Enum.GetNames(typeof(DatabaseType)).Length + 1;
```
Comment (lmalenfant) [inline src/Vts.Test/MonteCarlo/Extensions/DatabaseExtensionMethodsTests.cs:30]: Very cool!

File: src/Vts/Common/Range/DoubleRange.cs [hunk 2]
```diff
@@ -42,17 +42,14 @@ public DoubleRange()
         /// <returns>A double representing delta</returns>
         protected override double GetDelta()
         {
-            if (Count == 1)
+            if (Count != 1) return (Stop - Start) / (Count - 1);
+            if (Start != Stop)
             {
```
Comment (lmalenfant) [inline src/Vts/Common/Range/DoubleRange.cs:47]: Make this return value inline like the one above (LM)

File: src/Vts/Common/Range/FloatRange.cs [hunk 3]
```diff
@@ -42,17 +42,14 @@ public FloatRange()
         /// <returns>A float representing delta</returns>
         protected override float GetDelta()
         {
-            if (Count == 1)
+            if (Count != 1) return (Stop - Start) / (Count - 1F);
+            if (Start != Stop)
             {
```
Comment (lmalenfant) [inline src/Vts/Common/Range/FloatRange.cs:47]: Inline the return

File: src/Vts/Common/Range/IntRange.cs [hunk 4]
```diff
@@ -42,17 +42,14 @@ public IntRange()
         /// <returns>An integer representing delta</returns>
         protected override int GetDelta()
         {
-            if (Count == 1)
+            if (Count != 1) return (Stop - Start) / (Count - 1);
+            if (Start != Stop)
             {
```
Comment (lmalenfant) [inline src/Vts/Common/Range/IntRange.cs:47]: Inline the return

File: src/Vts/Common/Range/LongRange.cs [hunk 5]
```diff
@@ -42,17 +42,14 @@ public LongRange()
         /// <returns>A long representing delta</returns>
         protected override long GetDelta()
         {
-            if (Count == 1)
+            if (Count != 1) return (Stop - Start) / (Count - 1);
+            if (Start != Stop)
             {
```
Comment (lmalenfant) [inline src/Vts/Common/Range/LongRange.cs:47]: Inline the return

File: src/Vts/Common/Range/UIntRange.cs [hunk 6]
```diff
@@ -42,17 +42,14 @@ public UIntRange()
         /// <returns>An unsigned integer representing delta</returns>
         protected override uint GetDelta()
         {
-            if (Count == 1)
+            if (Count != 1) return (uint)((Stop - Start) / (Count - 1));
+            if (Start != Stop)
             {
```
Comment (lmalenfant) [inline src/Vts/Common/Range/UIntRange.cs:47]: Inline the return

File: src/Vts/MonteCarlo/Helpers/DetectorBinning.cs [hunk 7]
```diff
@@ -92,22 +88,18 @@ public static int WhichBin(double value, double[] binStops)
...
-            int bin = (int)Math.Floor((value - binStart) / binSize);
+            var bin = (int)Math.Floor((value - binStart) / binSize);
             if (bin > numberOfBins - 1)
             {
                 return -1;
             }
-            else
+
+            if (bin < 0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/DetectorBinning.cs:98]: This return could be inline

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 8]
```diff
@@ -386,22 +380,20 @@ public static double GetLimit(double factor)
         {
             if (lengthX == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:382]: Inline the return:
`if (lengthX == 0.0) return (center);`

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 9]
```diff
@@ -418,9 +410,9 @@ public static double GetLimit(double factor)
             double b,
             Random rng)
         {
-            if ((a == 0.0) && (b == 0.0))
+            if (a == 0.0 && b == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:414]: Inline the return

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 10]
```diff
@@ -453,34 +445,34 @@ public static double GetLimit(double factor)
             double beamDiaFWHM,
             Random rng)
         {
-            if ((a == 0.0) && (b == 0.0))
+            if (a == 0.0 && b == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:449]: Inline the return

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 11]
```diff
@@ -499,9 +491,9 @@ public static double GetLimit(double factor)
             double c,
             Random rng)
         {
-            if ((a == 0.0) && (b == 0.0) && (c == 0.0))
+            if (a == 0.0 && b == 0.0 && c == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:495]: Inline the return

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 12]
```diff
@@ -538,39 +530,39 @@ public static double GetLimit(double factor)
             double beamDiaFWHM,
             Random rng)
         {
-            if ((a == 0.0) && (b == 0.0) && (c == 0.0))
+            if (a == 0.0 && b == 0.0 && c == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:534]: Inline the return

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 13]
```diff
@@ -587,14 +579,17 @@ public static double GetLimit(double factor)
             double lengthY,
             Random rng)
         {
-            if ((lengthX == 0.0) && (lengthY == 0.0))
+            if (lengthX == 0.0 && lengthY == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:583]: Inline the return

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 14]
```diff
@@ -614,16 +609,16 @@ public static double GetLimit(double factor)
             double beamDiaFWHM,
             Random rng)
         {
-            if ((lengthX == 0.0) && (lengthY == 0.0))
+            if (lengthX == 0.0 && lengthY == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:613]: Inline the return

File: src/Vts/MonteCarlo/Helpers/DetectorBinning.cs [hunk 15]
```diff
@@ -52,28 +48,28 @@ public static int WhichBin(double value, int numberOfBins, double binSize, doubl
...
-            if ((value >= 0.0) && (value < binStops[0]))
+            if (value >= 0.0 && value < binStops[0])
                 return 0;
-            for (int i = 1; i < binStops.Count(); i++)
+            for (var i = 1; i < binStops.Count(); i++)
             {
-                if ((value >= binStops[i-1]) && (value < binStops[i]))
+                if (value >= binStops[i-1] && value < binStops[i])
                     return i;
             }
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/DetectorBinning.cs:74]: Inline the return

File: src/Vts/MonteCarlo/Helpers/DetectorBinning.cs [hunk 16]
```diff
@@ -92,22 +88,18 @@ public static int WhichBin(double value, double[] binStops)
         /// <returns>integer index of bin value is in</returns>
         public static int WhichBinExclusive(double value, int numberOfBins, double binSize, double binStart)
         {
-            int bin = (int)Math.Floor((value - binStart) / binSize);
+            var bin = (int)Math.Floor((value - binStart) / binSize);
             if (bin > numberOfBins - 1)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/DetectorBinning.cs:93]: Inline the return

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 17]
```diff
@@ -320,17 +314,17 @@ public static double GetLimit(double factor)
             double beamDiaFWHM,
             Random rng)
         {
-            if ((lengthX == 0.0) && (lengthY == 0.0) && (lengthZ == 0.0))
+            if (lengthX == 0.0 && lengthY == 0.0 && lengthZ == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:318]: Inline the return

File: src/Vts.Test/MonteCarlo/Sources/PointAndLineSources/LineSources/DirectionalLineSourceTests.cs [hunk 18]
```diff
@@ -19,17 +19,15 @@ public class DirectionalLineSourceTests
         [OneTimeSetUp]
         public void setup_validation_data()
```
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Sources/PointAndLineSources/LineSources/DirectionalLineSourceTests.cs:20]: Capitalize "Setup..."

File: src/Vts/MonteCarlo/Sources/PointAndLineSources/LineSources/DirectionalLineSource.cs [hunk 19]
```diff
@@ -178,15 +178,13 @@ protected override Direction GetFinalDirection(Position position)
                             new DoubleRange(0.0, Math.Abs(_thetaConvOrDiv)),
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Sources/PointAndLineSources/LineSources/DirectionalLineSource.cs:178]: Inline the return or add brackets?

File: src/Vts.Test/MonteCarlo/Sources/PointAndLineSources/PointSources/CustomPointSourceTests.cs [hunk 20]
```diff
@@ -20,17 +19,15 @@ public class CustomPointSourceTests
         [OneTimeSetUp]
         public void setup_validation_data()
```
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Sources/PointAndLineSources/PointSources/CustomPointSourceTests.cs:20]: Capitalize "Setup..."

File: src/Vts.Test/MonteCarlo/Sources/PointAndLineSources/PointSources/IsotropicPointSourceTests.cs [hunk 21]
```diff
@@ -19,17 +19,15 @@ public class IsotropicPointSourceTests
         [OneTimeSetUp]
         public void setup_validation_data()
```
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Sources/PointAndLineSources/PointSources/IsotropicPointSourceTests.cs:20]: Capitalize "Setup..."

File: src/Vts.Test/MonteCarlo/Sources/SurfaceEmittingFlatSources/Circular/CustomCircularSourceTests.cs [hunk 22]
```diff
@@ -20,17 +22,15 @@ public class CustomCircularSourceTests
         [OneTimeSetUp]
         public void setup_validation_data()
```
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Sources/SurfaceEmittingFlatSources/Circular/CustomCircularSourceTests.cs:23]: Capitalize "Setup..."

File: src/Vts/MonteCarlo/Sources/SurfaceEmittingFlatSources/Circular/DirectionalCircularSource.cs [hunk 23]
```diff
@@ -192,15 +192,12 @@ protected override Direction GetFinalDirection(Position position)
                             new DoubleRange(0.0, Math.Abs(_thetaConvOrDiv)),
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Sources/SurfaceEmittingFlatSources/Circular/DirectionalCircularSource.cs:192]: Inline the return or add brackets?

File: src/Vts.Test/MonteCarlo/Sources/SurfaceEmittingFlatSources/Elliptical/DirectionalEllipticalSourceTests.cs [hunk 24]
```diff
@@ -20,17 +20,15 @@ public class DirectionalEllipticalSourcesTests
         [OneTimeSetUp]
         public void setup_validation_data()
```
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Sources/SurfaceEmittingFlatSources/Elliptical/DirectionalEllipticalSourceTests.cs:21]: Capitalize "Setup..."

File: src/Vts/MonteCarlo/Sources/SurfaceEmittingFlatSources/Elliptical/DirectionalEllipticalSource.cs [hunk 25]
```diff
@@ -189,20 +189,17 @@ public class DirectionalEllipticalSource : EllipticalSourceBase
         /// <returns>new direction</returns>  
         protected override Direction GetFinalDirection(Position position)
         {
-            if ((_aParameter == 0.0) && (_bParameter == 0.0))
+            if (_aParameter == 0.0 && _bParameter == 0.0)
                 return (SourceToolbox.GetDirectionForGivenPolarAzimuthalAngleRangeRandom(
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Sources/SurfaceEmittingFlatSources/Elliptical/DirectionalEllipticalSource.cs:193]: Inline the return or add brackets

File: src/Vts.Test/MonteCarlo/Sources/SurfaceEmittingFlatSources/Rectangular/CustomRectangularSourceTests.cs [hunk 26]
```diff
@@ -19,18 +21,16 @@ public class CustomRectangularSourceTests
         [OneTimeSetUp]
         public void setup_validation_data()
```
Comment (hayakawa16) [inline src/Vts.Test/MonteCarlo/Sources/SurfaceEmittingFlatSources/Rectangular/CustomRectangularSourceTests.cs:22]: Capitalize "Setup..."

File: src/Vts/MonteCarlo/Sources/SurfaceEmittingFlatSources/Rectangular/DirectionalRectangularSource.cs [hunk 27]
```diff
@@ -188,22 +188,19 @@ public class DirectionalRectangularSource : RectangularSourceBase
         /// <returns>new direction</returns>  
         protected override Direction GetFinalDirection(Position position)
         {
-            
 
-            if ((_rectLengthX == 0.0) && (_rectWidthY == 0.0))
+
+            if (_rectLengthX == 0.0 && _rectWidthY == 0.0)
                 return (SourceToolbox.GetDirectionForGivenPolarAzimuthalAngleRangeRandom(
                             new DoubleRange(0.0, Math.Abs(_thetaConvOrDiv)),
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Sources/SurfaceEmittingFlatSources/Rectangular/DirectionalRectangularSource.cs:195]: Inline the return or add brackets

File: src/Vts/Common/Math/Convert.cs [hunk 28]
```diff
@@ -20,60 +20,60 @@ public static class Convert
         /// The method determines the phase given real and imaginary values
         /// </summary>
         /// <param name="real">The real component</param>
-        /// <param name="imag">The imaginary component</param>
+        /// <param name="imaginary">The imaginary component</param>
```
Comment (hayakawa16) [inline src/Vts/Common/Math/Convert.cs:23]: I'm okay with the filling our of word imaginary.  Keep changes.

File: src/Vts/MonteCarlo/Tissues/BoundedTissue.cs [hunk 29]
```diff
@@ -136,11 +127,9 @@ public new PhotonStateType GetPhotonDataPointStateOnExit(Position position)
             {
                 return PhotonStateType.PseudoReflectedTissueBoundary;
             }
-            if (Math.Abs(position.Z - ((LayerTissueRegion)_layers.Last()).ZRange.Start) < 1e-10)
-            {
-                return PhotonStateType.PseudoTransmittedTissueBoundary;
-            }
-            return PhotonStateType.PseudoBoundingVolumeTissueBoundary;
+            return Math.Abs(position.Z - ((LayerTissueRegion)_layers.Last()).ZRange.Start) < 1e-10
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/BoundedTissue.cs:130]: Inline the return or add brackets

File: src/Vts/MonteCarlo/Tissues/CaplessCylinderTissueRegion.cs [hunk 30]
```diff
@@ -61,21 +61,19 @@ public CaplessCylinderTissueRegion(Position center, double radius, double height
...
             if (inside < 0.9999999999 * Radius)
             {
                 return true;
             }
-            else if (inside > 1.00000000001 * Radius)
+            if (inside > 1.00000000001 * Radius)
             {
                 return false;
-            }
-            else  // on boundary means cylinder contains position
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/CaplessCylinderTissueRegion.cs:74]: Inline these returns lines 68 and 72

File: src/Vts/MonteCarlo/Tissues/EllipsoidTissueRegion.cs [hunk 31]
```diff
@@ -66,26 +66,25 @@ public class EllipsoidTissueRegion : ITissueRegion
...
+                             (Dz * Dz);
 
                 if (inside < 0.9999999999) // previous check  0.9999999
                 {
                     return true;
                 }
-                else if (inside > 1.00000000001) // previous check 1.0000001
+                if (inside > 1.00000000001) // previous check 1.0000001
                 {
                     return false;
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/EllipsoidTissueRegion.cs:82]: Inline the return

File: src/Vts/MonteCarlo/Tissues/EllipsoidTissueRegion.cs [hunk 32]
```diff
@@ -125,134 +124,130 @@ public bool RayIntersectBoundary(Photon photon, out double distanceToBoundary)
...
 
-                        // ckh fix 8/25/11: check if on boundary of ellipsoid
-                        if (distanceToBoundary < 1e-11)
-                        {
-                            return false;
-                        }
+                    // ckh fix 8/25/11: check if on boundary of ellipsoid
+                    if (distanceToBoundary < 1e-11)
+                    {
+                        return false;
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/EllipsoidTissueRegion.cs:223]: Inline the return

File: src/Vts/MonteCarlo/Tissues/InfiniteCylinderTissueRegion.cs [hunk 33]
```diff
@@ -58,25 +58,23 @@ public bool ContainsPosition(Position position)
...
             // GetDistanceToBoundary or code goes through cycles at cylinder boundary
             if (inside < (1 - 1e-9) * Radius)
             {
                 return true;
             }
-            else if (inside > (1 + 1e-9) * Radius)
+            if (inside > (1 + 1e-9) * Radius)
             {
                 return false;
             }
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/InfiniteCylinderTissueRegion.cs:73]: Inline the return

File: src/Vts/MonteCarlo/Tissues/MultiLayerWithSurfaceFiberTissue.cs [hunk 34]
```diff
@@ -202,26 +202,26 @@ public int GetNeighborRegionIndex(Photon photon)
...
                 return Math.Min(photon.CurrentRegionIndex + 1, Regions.Count - 1);
             }
-            else // pointed up
+
+            // pointed up
+            if (photon.CurrentRegionIndex != 1) // check if at surface
             {
-                if (photon.CurrentRegionIndex == 1) // check if at surface
+                if (photon.CurrentRegionIndex == 3) // in surfaceFiberRegion
                 {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/MultiLayerWithSurfaceFiberTissue.cs:210]: Can this return be on one line?

File: src/Vts/MonteCarlo/Tissues/MultiLayerWithSurfaceFiberTissue.cs [hunk 35]
```diff
@@ -202,26 +202,26 @@ public int GetNeighborRegionIndex(Photon photon)
...
+            else
+            {
+                if (_surfaceFiberRegion.ContainsPosition(photon.DP.Position))
                 {
-                    if (photon.CurrentRegionIndex == 3) // in surfaceFiberRegion
-                    {
-                        return 0;
-                    }
+                    return Regions.Count - 1; // return index of surfaceFiberRegion
                 }
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/MultiLayerWithSurfaceFiberTissue.cs:219]: Inline the return

File: src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs [hunk 36]
```diff
@@ -66,6 +65,8 @@ public class DatabaseWriterFactory
                 case DatabaseType.pMCDiffuseTransmittance:
                     return new PhotonDatabaseWriter(VirtualBoundaryType.pMCDiffuseTransmittance,
                         Path.Combine(filePath, outputName, "DiffuseTransmittanceDatabase"));
+                default:
+                    return null;
```
Comment (hayakawa16) [inline src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs:69]: This should return exception.  I will fix and add unit test.

File: src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs [hunk 37]
```diff
@@ -99,15 +100,19 @@ public class DatabaseWriterFactory
...
                 case DatabaseType.pMCDiffuseTransmittance:
                     return new CollisionInfoDatabaseWriter(VirtualBoundaryType.pMCDiffuseTransmittance,
                         Path.Combine(filePath, outputName, "CollisionInfoTransmittanceDatabase"),
-                        tissue.Regions.Count());
+                        tissue.Regions.Count);
+                case DatabaseType.DiffuseReflectance:
+                case DatabaseType.DiffuseTransmittance:
+                case DatabaseType.SpecularReflectance:
+                default:
+                    return null;
```
Comment (hayakawa16) [inline src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs:115]: This should return exception.  I will fix and write unit test.

File: src/Vts/MonteCarlo/Tissues/SingleInclusionTissue.cs [hunk 38]
```diff
@@ -63,21 +63,17 @@ public override int GetRegionIndex(Position position)
...
         {
             // first, check what region the photon is in
-            int regionIndex = photon.CurrentRegionIndex;
+            var regionIndex = photon.CurrentRegionIndex;
 
             // if we're in the layer region of the inclusion, could be on boundary of layer
             if ((regionIndex == _layerRegionIndexOfInclusion) && !Regions[_layerRegionIndexOfInclusion].OnBoundary(photon.DP.Position) )
             {
                 return _inclusionRegionIndex;
             }
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/SingleInclusionTissue.cs:72]: This if is really long so this one might look better with brackets, thoughts?

File: src/Vts/MonteCarlo/Tissues/SurfaceFiberTissueRegion.cs [hunk 39]
```diff
@@ -65,15 +65,15 @@ public bool ContainsPosition(Position position)
                 {
                     return true;
                 }
-                else if (inside > 1.00000000001) // prior check  1.0000001
+
+                if (inside > 1.00000000001) // prior check  1.0000001
                 {
                     return false;
                 }
-                else  // on boundary means SurfaceFiber contains position
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Tissues/SurfaceFiberTissueRegion.cs:72]: Inline the return

File: src/Vts/MonteCarlo/Factories/PhotonDatabaseFactory.cs [hunk 40]
```diff
@@ -35,6 +35,9 @@ public class PhotonDatabaseFactory
                 case VirtualBoundaryType.pMCDiffuseTransmittance: //pMC uses same exit db as regular post-processing
                     dbFilename = Path.Combine(filePath, "DiffuseTransmittanceDatabase");
                     break;
+                case VirtualBoundaryType.GenericVolumeBoundary:
+                case VirtualBoundaryType.Dosimetry:
+                case VirtualBoundaryType.BoundingCylinderVolume:
                 default:
                     return null;
```
Comment (hayakawa16) [inline src/Vts/MonteCarlo/Factories/PhotonDatabaseFactory.cs:42]: This should return exception.  I will fix and write unit test.

**Review by lmalenfant — COMMENTED** [review]
I made it through all the files, starting my edits

File: src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfRhoAndZLoader.cs [hunk 41]
```diff
@@ -135,19 +135,17 @@ public void InitializeFluorescentRegionArrays()
...
-                    if (inFluorescentTissue)
-                    {
-                        MapOfRhoAndZ[i, k] = 1;
-                        PDFOfRhoAndZ[i, k] = AOfRhoAndZ[i, k];
-                        TotalProb += AOfRhoAndZ[i, k];
-                        CDFOfRhoAndZ[i, k] += TotalProb;
-                    }
+                    if (!inFluorescentTissue) continue;
+                    MapOfRhoAndZ[i, k] = 1;
+                    PDFOfRhoAndZ[i, k] = AOfRhoAndZ[i, k];
```
Comment (hayakawa16) [inline src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfRhoAndZLoader.cs:140]: Change to "Pdf..." and "Cdf..."

File: src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfXAndYAndZLoader.cs [hunk 42]
```diff
@@ -142,22 +142,20 @@ public void InitializeFluorescentRegionArrays()
...
-                        if (inFluorescentTissue)
-                        {
-                            MapOfXAndYAndZ[i, j, k] = 1;
-                            PDFOfXAndYAndZ[i, j, k] = AOfXAndYAndZ[i, j, k];
-                            TotalProb += AOfXAndYAndZ[i, j, k];
-                            CDFOfXAndYAndZ[i, j, k] += TotalProb;
-                        }
+                        if (!inFluorescentTissue) continue;
+                        MapOfXAndYAndZ[i, j, k] = 1;
+                        PDFOfXAndYAndZ[i, j, k] = AOfXAndYAndZ[i, j, k];
```
Comment (hayakawa16) [inline src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfXAndYAndZLoader.cs:147]: Change to "Pdf..." and "Cdf...".

File: src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/FluorescenceEmissionAOfRhoAndZSource.cs [hunk 43]
```diff
@@ -123,58 +123,53 @@ public class FluorescenceEmissionAOfRhoAndZSource : FluorescenceEmissionSourceBa
...
-                            if (Loader.MapOfRhoAndZ[i, k] == 1)
-                            {
-                                if (rho < Loader.CDFOfRhoAndZ[i, k])
-                                {
-                                    // SHOULD I SAMPLE THIS W CYLINDRICAL (Y=0) OR CARTESIAN COORD?
-                                    xMidpoint = Loader.Rho.Start + i * Loader.Rho.Delta + Loader.Rho.Delta / 2;
-                                    yMidpoint = 0.0;
-                                    zMidpoint = Loader.Z.Start + k * Loader.Z.Delta + Loader.Z.Delta / 2;
+                            if (Loader.MapOfRhoAndZ[i, k] != 1) continue;
+                            if (rho >= Loader.CDFOfRhoAndZ[i, k]) continue;
```
Comment (hayakawa16) [inline src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/FluorescenceEmissionAOfRhoAndZSource.cs:138]: Change to Cdf..."

File: src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/FluorescenceEmissionAOfXAndYAndZSource.cs [hunk 44]
```diff
@@ -123,36 +123,31 @@ public class FluorescenceEmissionAOfXAndYAndZSource : FluorescenceEmissionSource
...
-                                    {
-                                        xMidpoint = Loader.X.Start + i * Loader.X.Delta + Loader.X.Delta / 2;
-                                        yMidpoint = Loader.Y.Start + j * Loader.Y.Delta + Loader.Y.Delta / 2;
-                                        zMidpoint = Loader.Z.Start + k * Loader.Z.Delta + Loader.Z.Delta / 2;
-                                        weight = 1.0;
-                                        return new Position(xMidpoint, yMidpoint, zMidpoint);
-                                    }
-                                }
+                                if (Loader.MapOfXAndYAndZ[i, j, k] != 1) continue;
+                                if (rho >= Loader.CDFOfXAndYAndZ[i, j, k]) continue;
```
Comment (hayakawa16) [inline src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/FluorescenceEmissionAOfXAndYAndZSource.cs:140]: Change to "Cdf..."

File: src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs (hunk 36 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs:69]: Fixed and unit test added.

File: src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs (hunk 37 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Factories/DatabaseWriterFactory.cs:115]: Fixed and unit test added for this and all other cases.

File: src/Vts/MonteCarlo/Tissues/SingleInclusionTissue.cs (hunk 38 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Tissues/SingleInclusionTissue.cs:72]: I kept brackets and used wrap line for "if".

File: src/Vts/MonteCarlo/Factories/PhotonDatabaseFactory.cs (hunk 40 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Factories/PhotonDatabaseFactory.cs:42]: Corrected to have actual cases return null and default return exception.

File: src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfRhoAndZLoader.cs (hunk 41 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfRhoAndZLoader.cs:140]: Can't make this change.  Array is property in class.

File: src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfXAndYAndZLoader.cs (hunk 42 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/AOfXAndYAndZLoader.cs:147]: Can't make this change, array is property in class.

File: src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/FluorescenceEmissionAOfRhoAndZSource.cs (hunk 43 above)
Comment (hayakawa16) [reply src/Vts/MonteCarlo/Sources/VolumetricSources/FluorescenceEmission/FluorescenceEmissionAOfRhoAndZSource.cs:138]: Can't make this change, array is property in class.

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 45]
```diff
@@ -289,16 +281,18 @@ public static double GetLimit(double factor)
             double lengthZ,
             Random rng)
         {
-            if ((lengthX == 0.0) && (lengthY == 0.0) && (lengthZ == 0.0))
+            if (lengthX == 0.0 && lengthY == 0.0 && lengthZ == 0.0)
             {
-                return (center);
+                return center;
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:286]: Inline return

File: src/Vts/MonteCarlo/Helpers/SourceToolbox.cs [hunk 46]
```diff
@@ -361,13 +352,13 @@ public static double GetLimit(double factor)
         {
             if (lengthX == 0.0)
             {
```
Comment (lmalenfant) [inline src/Vts/MonteCarlo/Helpers/SourceToolbox.cs:354]: Inline return

**Review by hayakawa16 — APPROVED** [review]

**Pull request merged**